content/blog/2024/12/how-to-migrate-from-idf-bootloader-to-mcuboot-using-esp-self-reflasher/index.md

```diff
@@ -0,0 +1,271 @@
+
+Save and exit the menuconfig.
+
+Build the example:
+
+```bash
+idf.py build
+```
+
+### Building and flashing the IDF application
+
+The IDF's `simple_ota_example` will be used as the hypothetical IDF application that will OTA to the Reflashing application. The steps are basically the same as described on its [documentation](https://github.com/espressif/esp-idf/tree/v5.1.4/examples/system/ota), with few changes.
+
+```bash
+cd <IDF_DIR>/examples/system/ota/simple_ota_example/
+```
+
+Set the target chip:
+
+```bash
+idf.py set-target esp32
+```
+
+Set the configurations for the example:
+
+```bash
+idf.py menuconfig
+```
+
+In the `Example Configuration` menu, set the `firmware upgrade url endpoint` with the host IP, port and the Reflasher application binary name (e.g. `http://192.168.0.100:8070/boot_swap_download.bin`):
+
+```
+http://<HOST_IP>:<HOST_PORT>/<reflashing_application_name>.bin
+```
+
+In the `Example Connection Configuration` menu, again set the preferred network connection settings, such as Wi-Fi SSID and Password.
+
+Only for illustration purposes, this guide is using simple HTTP requests, so enable the `Allow HTTP for OTA` option in the `Component config -> ESP HTTPS OTA`.
+
+---
+If the `boot_swap_embedded_example` was used, define a custom partition table in `Partition table` menu. Create the `partitions.csv` file with the following content:
+
+```
+# Name,     Type, SubType,     Offset,          Size, Flags
+nvs,        data, nvs,               ,        0x4000,
+otadata,    data, ota,               ,        0x2000,
+phy_init,   data, phy,               ,        0x1000,
+factory,    app,  factory,    0x10000,      0x180000,
+ota_0,      app,  ota_0,             ,      0x180000,
+```
+
+---
+
+Save and exit the menuconfig.
+
+Copy the Reflashing application binary that was build in the previous section to the directory where the HTTP server will run:
+
+```bash
+cp <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_<MODE>_example/build/boot_swap_<MODE>_example.bin <HTTP_SERVER_DIR>
+```
+
+Open a new bash and start the HTTP server for the OTA application. Here is a suggestion for creating the HTTP server using Python:
+
+```bash
+cd <HTTP_SERVER_DIR>
+sudo python -m http.server -b <HOST_IP> <HOST_PORT>
+# e.g. sudo python -m http.server -b 192.168.0.100 8070
+```
+
+---
+If the `boot_swap_download_example` was used, copy the target reflashing images to the <HTTP_SERVER_DIR> or open another bash and start the HTTP server from where the reflashing images will be downloaded (watch for the <PORT> that cannot be the same):
+
+```bash
+cd <IDF_DIR>/components/esp-self-reflasher/examples/boot_swap_download_example/example_bin_dir/
+sudo python -m http.server -b <HOST_IP> <HOST_PORT>
+# e.g. sudo python -m http.server -b 192.168.0.100 8071
+```
+
+---
+
+Finally build and flash the OTA application:
+
+```bash
+idf.py -p <SERIAL_PORT> flash monitor
+# e.g. idf.py -p /dev/ttyUSB0 flash monitor
+```
+
+## Conclusion
```

Comment:
Please add a paragraph about what has been done in the course of this article.

This will naturally lead to the existing paragraph that gives final suggestions before considering the migration.